src/Validation/Validator.php

```diff
@@ -525,6 +527,14 @@ protected function _fieldIsEmpty($data)
         if (empty($data) && $data !== '0' && $data !== false && $data !== 0 && $data !== 0.0) {
             return true;
         }
+        $isArray = is_array($data);
+        if ($isArray && (isset($data['year']) || isset($data['hour']))) {
+            $value = implode('', $data);
+            return strlen($value) === 0;
+        }
+        if ($isArray && isset($data['name'], $data['type'], $data['tmp_name'], $data['error'])) {
+            return (int)$data['error'] === UPLOAD_ERR_NO_FILE;
```

Comment:
I think a file being too large shouldn't count as empty. They did upload a file and that should be covered by other validation rules. allowEmpty() should only be covering if the file input is submitted empty imo.